src/logged-evaluation.nix

```diff
@@ -122,9 +122,12 @@ let
 
             # Export IN_NIX_SHELL to trick various Nix tooling to export
             # shell-friendly variables
-
             export IN_NIX_SHELL=impure
 
+            # Export IN_LORRI_SHELL to differentiate between `lorri shell` and
+            # `nix-shell`
+            export IN_LORRI_SHELL=${src}
```

Comment:
This needs to be escaped via `lib.escapeShellArg`. I think it’s gonna be a lot of work importing `lib` into this file, so let’s just inline the definition from `nixpkgs/lib/strings.nix` into this file:

```
  escapeShellArg = arg: "'${builtins.replaceStrings ["'"] ["'\\''"] (toString arg)}'";
```

Reply:
What about spaces? Should we worry about those? As it stands, having a folder `asd fg'h` will break e.g. `stat $IN_LORRI_SHELL`.

Reply:
`escapeShellArg` will do the right thing for both spaces and single quotes. It wraps everything in single quotes, taking care of spaces, and then it leaves the single quote every time it encounters one, to prevent escaping issues.

`ab c'de` -> `'ab c'\''de'`

Bash implicitly concatenates two strings if they have no spaces between each other, so both single-quoted strings are merged into one string.